mm2src/mm2_main/src/lp_swap/taker_swap_v2.rs

```diff
@@ -1798,7 +1811,7 @@ impl<MakerCoin: MmCoin + MakerCoinSwapOpsV2, TakerCoin: MmCoin + TakerCoinSwapOp
             funding_time_lock: state_machine.taker_funding_locktime(),
             payment_time_lock: state_machine.taker_payment_locktime(),
             maker_pubkey: &self.negotiation_data.taker_coin_htlc_pub_from_maker,
-            taker_secret: state_machine.taker_secret.as_slice(),
+            taker_secret: &state_machine.taker_secret.take(),
```

Comment:
do you really mean to take out the value? why not `as_slice` and `try_into`?

Reply:
I believe it's better to use impl_hash macro method that doesn't return an error.
![image](https://github.com/user-attachments/assets/5bd8b6ce-9644-4b08-8ca7-2982d489f24b)

```rust
impl_hash!(H256, 32);
```
![image](https://github.com/user-attachments/assets/faf185fb-41a9-4d89-95c0-d5747008bb6a)

Reply:
ok I didn't check that... the method name `take` is confusing